lib/orogen/gen/templates.rb

```diff
@@ -9,10 +9,22 @@ class << self
 	    attr_reader :templates
 	end
 
-        # Returns the directory where Orogen's lib part sits (i.e. where
-        # autobuild.rb and autobuild/ are)
+        # Returns the directory where oroGen's lib part sits, i.e.
+        # base_dir is defined by the directory where orogen.rb resides
         def self.base_dir
```

Comment:
I'm not sure I'm following. `base_dir` was returning the place of the contents of `lib/` isn't it ? Isn't that the same than the contents of `vendor_ruby`, i.e. `File.join(base_dir, 'orogen')` would be true in both setups. What am I missing ?

The problem as far as I understand it was that the templates were trying to go *out* of lib/ which is not possible. The fix you've applied (moving them into lib/) is definitely the right one.

> Also, if I can assume that there are no further references relative to and going outside of lib/ (e.g. by plugins) then this fix is obsolete.

That's definitely what we should enforce that.